js/noms/src/value-store-test.js

```diff
@@ -87,6 +90,31 @@ suite('ValueStore', () => {
     await vs.close();
   });
 
+  test('read nonexistince caching', async () => {
+    const bs = new BatchStoreAdaptor(new MemoryStore());
+    const vs = new ValueStore(bs, 1e6);
+
+    const hash = notNull(Hash.parse('rmnjb8cjc5tblj21ed4qs821649eduie'));
+    const v1 = await vs.readValue(hash);
+    assert.equal(null, v1);
+    (bs: any).get = () => { throw new Error(); };
+    const v2 = await vs.readValue(hash);
+    assert.equal(null, v2);
+    await vs.close();
+  });
+
+  test('write clobbers cached nonexistence', async () => {
+    const vs = new ValueStore(new BatchStoreAdaptor(new MemoryStore()), 1e6);
```

Comment:
You should be able to use a much smaller cache here, you're only storing "hello".